lms/static/js/spec/search/search_spec.js

```diff
@@ -0,0 +1,513 @@
+            expect(onNext).toHaveBeenCalled();
+        });
+
+        it('shows a spinner when loading more results', function () {
+            this.collection.totalCount = 123;
+            this.collection.hasNextPage = function () { return true; };
+            this.listView.render();
+            this.listView.loadNext();
+            expect(this.listView.$el.find('a.search-load-next .icon')[0]).toBeVisible();
+            this.listView.renderNext();
+            expect(this.listView.$el.find('a.search-load-next .icon')[0]).toBeHidden();
+        });
+
+    });
+
+
+    describe('SearchRouter', function () {
+
+        beforeEach(function () {
+            this.router = new SearchRouter();
+        });
+
+        it ('has a search route', function () {
+            expect(this.router.routes['search/:query']).toEqual('search');
+        });
+
+    });
+
+
+    describe('SearchApp', function () {
+
+        beforeEach(function () {
+            loadFixtures('js/fixtures/search_form.html');
+            appendSetFixtures(
+                '<section id="courseware-search-results" data-course-name="Test Course"></section>' +
+                '<section id="course-content"></section>'
+            );
+            TemplateHelpers.installTemplates([
+                'templates/courseware_search/search_item',
+                'templates/courseware_search/search_list',
+                'templates/courseware_search/search_loading',
+                'templates/courseware_search/search_error'
+            ]);
+
+            this.server = Sinon.fakeServer.create();
+            this.server.respondWith([200, {}, JSON.stringify({
+                total: 1337,
+                access_denied_count: 12,
+                results: [{
+                    data: {
+                        location: ['section', 'subsection', 'unit'],
+                        url: '/some/url/to/content',
+                        content_type: 'text',
+                        excerpt: 'this is a short excerpt'
+                    }
+                }]
+            })]);
+
+            Backbone.history.stop();
+            this.app = new SearchApp('a/b/c');
+
+            // start history after the application has finished creating
+            //  all of its routers
+            Backbone.history.start();
+        });
+
+        afterEach(function () {
+            this.server.restore();
+        });
+
+        it ('shows loading message on search', function () {
+            $('.search-field').val('search string');
+            $('.search-button').trigger('click');
+            expect($('#course-content')).toBeHidden();
+            expect($('#courseware-search-results')).toBeVisible();
+            expect($('#courseware-search-results')).not.toBeEmpty();
+        });
+
+        it ('performs search', function () {
+            $('.search-field').val('search string');
+            $('.search-button').trigger('click');
+            this.server.respond();
```

Comment:
This test unfortunately causes a redirect to the query page. This makes it hard to run the debug the tests in a browser as the browser changes to this URL:

    suite/search/search/search/search/query

Please change it so that the submit doesn't actually happen during the test.